examples/server/utils.hpp

```diff
@@ -325,10 +326,16 @@ inline std::string format_chat(const struct llama_model * model, const std::stri
             throw std::runtime_error("Missing 'content' (ref: https://github.com/ggerganov/llama.cpp/issues/8367)");
         }
 
-        chat.push_back({role, content});
+        if (tmpl == "custom") {
```

Comment:
Maybe we can do this too?

```suggestion
        bool is_custom = !prefix.empty() || !suffix.empty();
        if (is_custom) {
```

Reply:
The UI logic and underlay logic should be decoupled. If you have `"custom"` here, it become redundant because you already had `bool is_custom = !prefix.empty() || !suffix.empty()`, so kinda 2 variables control the same state.